### PR TITLE
Feature/impl user password expiration check

### PR DESCRIPTION
**Pull request by dtsarenko**

### CODE REVIEW

File: src/test/java/com/icthh/xm/uaa/security/UaaAuthenticationProviderIntTest.java [hunk 1]
```diff
@@ -143,6 +148,20 @@ public void testFirstLdapAuth() {
         commonAsserts(TEST_USER, DEFAULT_USER_ROLE_KEY, authentication);
     }
 
+    @Test
+    public void testCheckPasswordExpirationSuccess() {
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/uaa/security/UaaAuthenticationProviderIntTest.java]: add unsuccesful test

Comment (dtsarenko) [reply src/test/java/com/icthh/xm/uaa/security/UaaAuthenticationProviderIntTest.java]: added

File: src/main/java/com/icthh/xm/uaa/domain/properties/TenantProperties.java [hunk 2]
```diff
@@ -175,4 +175,7 @@ public Boolean getCreateAccountAutomatically() {
             private Map<String, String> mapping;
         }
     }
+
+    @JsonProperty("passwordExpirationPeriod")
+    private Integer passwordExpirationPeriod = -1;
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/domain/properties/TenantProperties.java]: maybe it worth to put this property in `Security` container?

File: src/main/java/com/icthh/xm/uaa/security/UaaAuthenticationProvider.java [hunk 3]
```diff
@@ -59,9 +71,25 @@ private AuthenticationProvider getProvider(Authentication authentication) {
...
+    private void checkPasswordExpiration(Authentication authentication) {
+        DomainUserDetails domainUserDetails = (DomainUserDetails) authentication.getPrincipal();
+        User user = userService.getUser(domainUserDetails.getUserKey());
+        LocalDate updatePasswordDate = user.getUpdatePasswordDate().atZone(UTC).toLocalDate();
+        Integer expirationPeriod = tenantPropertiesService.getTenantProps().getPasswordExpirationPeriod();
+        LocalDate currentDate = LocalDate.now();
+        log.info("check password expiration, passwordUpdateDate: {}, currentDate: {}, expirationPeriod: {}",
+            updatePasswordDate, currentDate, expirationPeriod);
+        long period =  DAYS.between(updatePasswordDate, currentDate);
+        if (period > expirationPeriod && expirationPeriod > 0) {
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/security/UaaAuthenticationProvider.java]: Suggest to separate IF to check `expirationPeriod > 0` and perform all calculations and repository calls only if need:
```
private void checkPasswordExpiration(Authentication authentication) {
        Integer expirationPeriod = tenantPropertiesService.getTenantProps().getPasswordExpirationPeriod();
        if(expirationPeriod > 0) {
                User user = userService.getUser(domainUserDetails.getUserKey());
                  ....
                 if (period > expirationPeriod){
                            throw new CredentialsExpiredException("Password expiration period is over, please change password");
                 }
        }   

}
```

File: src/main/resources/config/liquibase/changelog/20190624165700_add_update_password_date_field.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet id="20190624165700" author="dtsarenko">
+        <addColumn tableName="jhi_user">
+            <column name="update_password_date" type="timestamp" defaultValueDate="${now}"/>
```
Comment (amedvedchuk) [inline src/main/resources/config/liquibase/changelog/20190624165700_add_update_password_date_field.xml]: setting to default value may lead to filling this column for all tenants with currdate. Please check if it won't affect their business logic

**Pull request merged**